staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -495,3 +516,42 @@ func isExpiredError(err error) bool {
 	// check when we fully drop support for Kubernetes 1.17 servers from reflectors.
 	return apierrors.IsResourceExpired(err) || apierrors.IsGone(err)
 }
+
+// backoffManager manages jittered exponential back off with a cap and a reset period. Exponential back off will stop
+// to increase at the given max, and timer will reset after not backing off for a while
+type backoffManager struct {
+	backoff              *wait.Backoff
+	backoffTimer         *time.Timer
+	lastBackoffStart     time.Time
+	initialBackoff       time.Duration
+	backoffResetDuration time.Duration
+}
+
+func newBackoffManager(initBackoff, maxBackoff, resetDuration time.Duration, backoffFactor, jitter float64) *backoffManager {
+	return &backoffManager{
+		backoff: &wait.Backoff{
+			Duration: initBackoff,
+			Factor:   backoffFactor,
+			Jitter:   jitter,
+
+			// the current impl of wait.Backoff returns Backoff.Duration once steps are used up, which is not
+			// what we ideally need here, we set it to max int and assume we will never use up the steps
+			Steps: math.MaxInt32,
+			Cap:   maxBackoff,
+		},
+		backoffTimer:         time.NewTimer(0),
+		initialBackoff:       initBackoff,
+		lastBackoffStart:     time.Now(),
+		backoffResetDuration: resetDuration,
+	}
+}
+
+func (b *backoffManager) Backoff() *time.Timer {
+	if time.Now().Sub(b.lastBackoffStart) > b.backoffResetDuration {
```

Comment:
Use the [clock](https://godoc.org/github.com/kubernetes/utils/clock) package to make this testable. (Test code uses the [fake](https://godoc.org/github.com/kubernetes/utils/clock/testing#FakeClock)).